Add configurable GRAVITY_TMPDIR variable into setupVars

kot0dama opened the pull request:
**What does this PR aim to accomplish?:**

This allows users to set up a Gravity updates temporary directory other than /tmp.
Indeed, some systems have a low amount of memory and some users might be reluctant to allocate too much memory to /tmp on these systems. You can look up the Internet for threads where people ask for a way to configure it.

**How does this PR accomplish the above?:**

I've added support for a GRAVITY_TMPDIR into gravity.sh, defaulting to /tmp if not configured, or if configured directory does not exist/is not a directory. 

I've also added a default in the setup tool, I'm not sure that is the way to go though.

This way, users will be able to edit /etc/pihole/setupVars.conf and set/update GRAVITY_TMPDIR to the desired value.
This is probably unnecessary to add it as a step in the configurator though, but feel free to add it if it makes sense.

**Link documentation PRs if any are needed to support this PR:**

None

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

dschaper commented on gravity.sh:
```diff
@@ -52,6 +52,20 @@ else
+if [ -z "${GRAVITY_TMPDIR}" ]; then
+  GRAVITY_TMPDIR="/tmp"
+else
+  if [ ! -e "${GRAVITY_TMPDIR}" ]; then
+    echo -e "  ${COL_LIGHT_RED}Gravity temporary directory does not exist, falling back to /tmp. ${COL_NC}"
+    GRAVITY_TMPDIR="/tmp"
+  fi
```
Not sure if there's a need to check for both existence and if is a directory. Should be able to simplify to just checking if the path is a directory. `-d` checks both existence and directory-ness. 

Might want to add in `-w` to check if the directory is writeable as well.

kot0dama replied:
Let's keep it simple indeed, updated the code accordingly.

dschaper commented on automated install/basic-install.sh:
```diff
@@ -1845,6 +1845,7 @@ finalExports() {
     addOrEditKeyValPair "${setupVars}" "DNS_FQDN_REQUIRED" "${DNS_FQDN_REQUIRED:-true}"
     addOrEditKeyValPair "${setupVars}" "DNS_BOGUS_PRIV" "${DNS_BOGUS_PRIV:-true}"
     addOrEditKeyValPair "${setupVars}" "DNSMASQ_LISTENING" "${DNSMASQ_LISTENING:-local}"
+    addOrEditKeyValPair "${setupVars}" "GRAVITY_TMPDIR" "${GRAVITY_TMPDIR:-/tmp}"
```
Don't think we really need this here, this function is meant to update `setupVars.conf` configuration lines if the values they point to were changed by the installation script.

kot0dama replied:
All good, removed that :)

dschaper commented on gravity.sh:
```diff
@@ -52,6 +52,20 @@ else
 fi
 
+# Set up tmp dir in case it's not configured
+if [ -z "${GRAVITY_TMPDIR}" ]; then
+  GRAVITY_TMPDIR="/tmp"
```
```suggestion
: "${GRAVITY_TMPDIR:=/tmp}"
```

See https://github.com/pi-hole/pi-hole/blob/d86b325dfe4d39b7bc523672bb22f1bdbd12265f/automated%20install/basic-install.sh#L38

```sh
Viking:~$ : "${RED:=blue}"
Viking:~$ echo $RED
blue
Viking:~$ unset RED
Viking:~$ RED='green'
Viking:~$ : "${RED:=blue}"
Viking:~$ echo $RED
green
```

dschaper commented on gravity.sh:
```diff
@@ -52,6 +52,20 @@ else
   exit 1
 fi
 
+# Set up tmp dir in case it's not configured
```
```suggestion
# Set up tmp dir variable in case it's not configured
```

kot0dama replied:
Thanks, done !

kot0dama commented:
Thanks for reviewing it, I think I've addressed all comments, let me know if there's any issue.
I've pushed another commit but then I can also squash them and force push.

Thank you!

dschaper commented:
Thanks, can you squash and force please?

kot0dama commented:
> Thanks, can you squash and force please?

Done and merged development branch again ;)

dschaper commented:
I don't know if this solves https://github.com/pi-hole/pi-hole/issues/5168 but it's very relevant to that.

kot0dama commented:
Cool, that was fast !
Thank you !
